### PR TITLE
Support rendering variables in footnotes

### LINKED ISSUE #379: How to use template in footnotes
## Describe your problem
I want to use template in footnotes, but it not work

## More details about your problem
There are footnotes in my DOCX document, and I need to replace the text in the footnotes with templates.  I added a footnote like {{var}}, but the final render docX does not replace var correctly, just shows {{var}} in the footnote

### PR DESCRIPTION
**Pull request by bartbroere**
This PR enables rendering variables that appear in footnotes.

Resolves #379

### CODE REVIEW

File: docxtpl/template.py [hunk 1]
```diff
@@ -351,6 +351,19 @@ def render_properties(
...
+    def render_footnotes(
+            self, context: Dict[str, Any], jinja_env: Optional[Environment] = None
+    ) -> None:
+        if jinja_env is None:
+            jinja_env = Environment()
+
+        for section in self.docx.sections:
+            for part in section.part.package.parts:
+                if part.content_type == 'application/vnd.openxmlformats-officedocument.wordprocessingml.footnotes+xml':
+                    xml = self.patch_xml(part.blob.decode('utf8'))
```
Comment (AsterisMono) [inline docxtpl/template.py:363]: `part.blob` somehow becomes `str` in the next loop body if we have multiple footnotes in the template.

```
Traceback (most recent call last):
  File "/Users/cmiki/Projects/sq-docxgen/src/templating/main.py", line 76, in <module>
    tpl.render(context, jinja_env)
  File "/Users/cmiki/Projects/python-docx-template/docxtpl/template.py", line 499, in render
    self.render_footnotes(context, jinja_env)
  File "/Users/cmiki/Projects/python-docx-template/docxtpl/template.py", line 363, in render_footnotes
    xml = self.patch_xml(part.blob.decode('utf8'))
                         ^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'decode'. Did you mean: 'encode'?
```

here's a duct-tape approach (which works on my template):
```suggestion
                    xml = self.patch_xml(part.blob.decode('utf-8') if type(part.blob) is bytes else part.blob)
```

Comment (bartbroere) [reply docxtpl/template.py:363]: Thanks for this, I believe my test only processes one footnote, so it didn't catch this.

For now I will apply your suggestion, but replace the `type(part.blob) is bytes` pattern with `isinstance(part.blob, bytes)`.

I will also try to narrow down what causes this in the `python-docx` library, which I believe might be causing this type confusion.

Comment (AsterisMono) [reply docxtpl/template.py]: Awesome. Thank you for this patch!

**Comment by elapouya** [comment]
Thank you for your PR. I will package it soon.

**Pull request merged**